src/loaders.rs

```diff
@@ -79,16 +119,30 @@ impl FileSystemLoader {
     }
 }
 
-pub struct AppDirsLoader {}
+pub struct AppDirsLoader {
+    encoding: &'static Encoding,
+}
 
 impl AppDirsLoader {
+    pub fn new(encoding: &'static Encoding) -> Self {
+        Self { encoding }
+    }
+
     fn get_template(
         &self,
-        _py: Python<'_>,
-        _template_name: &str,
-        _engine: &EngineData,
+        py: Python<'_>,
+        template_name: &str,
+        engine: &EngineData,
     ) -> Result<PyResult<Template>, LoaderError> {
-        todo!()
+        let dirs_result = get_app_template_dirs(py, "templates");
+        let dirs = match dirs_result {
+            Ok(dirs) => dirs,
+            Err(e) => Vec::new(),
+        };
+
+        let filesystem_loader = FileSystemLoader::new(dirs, self.encoding);
```

Comment:
I see that `FileSystemLoader::new` takes a `Vec<String>` and I assume that's why you made `get_app_template_dirs` return `Result<Vec<String>, PyErr>`. But `FileSystemLoader` stores `dirs` as a `Vec<PathBuf>`, which would be a more natural return type from `get_app_template_dirs`. For this reason, I think we should add a second constructor method to `FileSystemLoader` that takes a `Vec<PathBuf>` instead of `Vec<String>`.